app/models/commitment.rb

```diff
@@ -231,6 +233,19 @@ def self.pages(item_count)
     (item_count / @items_per_page.to_d).ceil
   end
 
+  def progress_documents_attributes=(attrs)
+    attrs.each do |attr|
+      next unless attr[:document].blank?
+
+      # If there's no document we want to destroy the record, and we
+      # need to remove the document value because it raises an error
+      # if it isn't a valid signed_id
+      attr.except!(:document)
+      attr[:_destroy] = true
+    end
+    super
+  end
```

Comment:
This may also be unnecessary now, since we're falling back to `addDestroyKeys` and should be removing everything else until submission, but we should confirm this in the params